joblibspark/backend.py

```diff
@@ -90,8 +90,11 @@ def effective_n_jobs(self, n_jobs):
             # n_jobs=-1 means requesting all available workers
             n_jobs = max_num_concurrent_tasks
         if n_jobs > max_num_concurrent_tasks:
-            n_jobs = max_num_concurrent_tasks
-            warnings.warn("limit n_jobs to be maxNumConcurrentTasks in spark: " + str(n_jobs))
+            warnings.warn("User-specified n_jobs ({n}) is greater than "
+                          "max_num_concurrent_tasks ({c}), "
```

Comment:
`max_num_concurrent_tasks` is an internal variable name. Just say `is greater than the max number of concurrent tasks (?) this cluster can run now. If dynamic allocation is enabled for the cluster, you might see more executors allocated.`